lib/src/Base/Stat/CorrelationAnalysis.cxx

```diff
@@ -32,32 +32,30 @@ BEGIN_NAMESPACE_OPENTURNS
 CorrelationAnalysis::CorrelationAnalysis() {}
 
 /* Compute the Pearson correlation coefficient between the component number index of the input sample and the 1D outputSample */
-Scalar CorrelationAnalysis::PearsonCorrelation(const Sample & inputSample,
-    const Sample & outputSample,
-    const UnsignedInteger index)
+Point CorrelationAnalysis::PearsonCorrelation(const Sample & inputSample,
+                                              const Sample & outputSample)
 {
-  if (index >= inputSample.getDimension()) throw InvalidArgumentException(HERE) << "Error: given index out of bound";
   if (outputSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
   if (inputSample.getSize() != outputSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
-  const UnsignedInteger size = inputSample.getSize();
-  Sample pairedSample(size, 2);
-  for (UnsignedInteger i = 0; i < size; ++i)
+  Sample pairedSample(inputSample);
+  pairedSample.stack(outputSample);
+  const UnsignedInteger dimension = inputSample.getDimension();
+  Point result(dimension);
+  CorrelationMatrix correlation(pairedSample.computePearsonCorrelation());
```

Comment:
@jschueller Looks like too many correlations are computed here. You compute a full matrix to extract only its last row!

Reply:
ok